cmd/loop: update urfave/cli to v3, auto-generate loop CLI reference manual and man file

starius opened the pull request:
## Summary

  - Upgraded the Loop CLI to `urfave/cli/v3`, adjusted every command/action to the new API
  - Hardened user input across commands by parsing numeric flags with their native helpers, rejecting negative amounts, and keeping static-address labels consistent.
  - Added `loop man` / `loop markdown` helpers plus Make/Ci steps that regenerate manpages/markdown and verify docs stay up to date.
  - Hid the development-only `forceautoloop` command from end users.

## Testing

  - `make build`
  - `loop man`
  - `loop markdown`
  - `make docs-check`
  - Manually testing `loop` to make sure it works properly

bhandras commented on cmd/loop/markdown_tabular.md.gotmpl:
```diff
@@ -0,0 +1,80 @@
+{{ define "flags" }}
```
This is really cool!

bhandras reviewed (approved):
Thank you! LGTM!!

starius commented:
Rebased

starius commented:
Updated static loop in usage:
```
USAGE:
   loop static in [amt] [--all | --utxo xxx:xx]
```

Updated docs for `--fast` flag and fixed `make docs-check` so it actually catches if docs is outdated.

starius commented:
Fixed documentation generation to put paths like `~/.loop` instead of expanded paths like `/home/user/.loop` which were breaking reproducibility.

CI was catching this with diffs like this:

```diff
-\fB--loopdir\fP="": path to loop's base directory (default: /home/user/.loop)
+\fB--loopdir\fP="": path to loop's base directory (default: /home/runner/.loop)
```
